fdmi/plugins/fdmi_app

```diff
@@ -118,13 +216,21 @@ def process_fdmi_record(kv_record, p):
     # Check de-dup records
     kv_i = kv_records.get(keystr)
     if kv_i is not None:
-        print('dup: {}'.format(kv), file=sys.stdout, flush = True)
-        return
+        #print('dup: {}'.format(kv), file=sys.stdout, flush = True)
+        #send ACK to plugin immediately beause it is a duplicated one
+        send_ack_to_plugin("++++++++++", p, rec_id)
+        pass
     else:
-        print('new: {}'.format(kv), file=sys.stdout, flush = True)
+        #print('new: {}'.format(kv), file=sys.stdout, flush = True)
         kv_records[keystr] = kv
         keystr_LRU.append(keystr)
-        return
+        #send FDMI event to kafka asynchronously. When this FDMI event is
+        #received by our "Acker" we will send it to plugin for acknowledgement.
+        kafka_producer_send(kv)
```

Comment:
Is kafka_producer_send is responsible to write to Kafka and then send ack to FDMI?  What will happen if it can not write to Kafka or what if it gets an error?

Reply:
This is the happy path.
More robust error handling should be tested and added.

If kafka server is not available or writing to kafka fails, we will not send ACK back to FDMI source.
FDMI will resend the FDMI events (This is a TODO)

Reply:
ok